File: 1346-MaximizeTheTopmostElementAfterKMoves/1346-MaximizeTheTopmostElementAfterKMoves.py
# Last updated: 07/04/2026, 11:36:42
class Solution:
    def maximumTop(self, nums: List[int], k: int) -> int:
        ans = -1
        n = len(nums)
        if ( k ==0 and n !=0 ) : return nums[0]
        if ( n ==1) : return nums[0] if k%2 ==0 else ans
        if (k ==1 ): return nums[1]
        if ( k  < n ) : ans = max ( ans, nums[k])
        
        
        ans = max(ans,max(nums[:k-1]))


        return ans
        
        
        

File: 1346-MaximizeTheTopmostElementAfterKMoves/1346-MaximizeTheTopmostElementAfterKMoves.cpp
// Last updated: 07/04/2026, 11:36:46
class Solution {
public:
    int maximumTop(vector<int>& nums, int k) {

        int n = nums.size();
        if ( k ==0 && n!= 0) return nums[0];
        if( n ==1 &&( k%2 ==1)) return -1;
        if ( k ==1) return nums[1];
        
        int  i = 0 ;
        int ans = 0 ; 
        
        if ( k < n){
            ans = nums[k];
            
        }
        --k;


        
        while(k-- && i <n){
            ans = max(ans,nums[i++]);
        }
    
        return ans;
        

        
    }
};